Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForestState.java

```diff
@@ -18,52 +18,55 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import lombok.Data;
+
 import com.amazon.randomcutforest.executor.Sequential;
-import com.amazon.randomcutforest.sampler.CompactSamplerData;
+import com.amazon.randomcutforest.sampler.CompactSamplerState;
 import com.amazon.randomcutforest.sampler.Weighted;
 import com.amazon.randomcutforest.store.PointStoreDoubleData;
-import com.amazon.randomcutforest.tree.TreeData;
+import com.amazon.randomcutforest.tree.CompactRandomCutTreeState;
 
 /**
  * A class that encapsulates most of the data used in a RandomCutForest such
  * that the forest can be serialized and deserialized.
  */
-public class ForestState {
-
-    public String typeOfForest = null;
+@Data
+public class RandomCutForestState {
 
-    public long entreesSeen;
+    private long entreesSeen;
 
-    public double lambda;
+    private double lambda;
 
-    public int numberOfTrees;
+    private int numberOfTrees;
 
-    public int sampleSize;
+    private int sampleSize;
 
-    public int dimensions;
+    private int dimensions;
 
-    public int outputAfter;
+    private int outputAfter;
 
-    public boolean storeSequenceIndexesEnabled;
+    private boolean storeSequenceIndexesEnabled;
 
-    public boolean compactEnabled;
+    private boolean compactEnabled;
 
-    public boolean centerOfMassEnabled;
+    private boolean centerOfMassEnabled;
 
-    public boolean parallelExecutionEnabled;
+    private boolean parallelExecutionEnabled;
 
-    public int threadPoolSize;
+    private int threadPoolSize;
 
     public boolean saveTreeData;
 
+    private Object coordinatorState;
+    private List<Object> componentModelStates;
+
     public PointStoreDoubleData pointStoreDoubleData;
 
-    public ArrayList<CompactSamplerData> compactSamplerData;
+    public ArrayList<CompactSamplerState> compactSamplerData;
 
     public ArrayList<List<Sequential<double[]>>> sequentialSamplerData;
 
     public ArrayList<List<Weighted<double[]>>> smallSamplerData;
 
-    public ArrayList<TreeData> treeData;
-
+    public ArrayList<CompactRandomCutTreeState> treeData;
```

Comment:
Can we make all of these private as well?

Reply:
That's the end goal. I'm pretty sure I left this because existing code that I want to replace with the mapper classes expects it to be public and I didn't want to increase the scope of this PR. We can make this change after we remove the old code.